ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +53,42 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    # We need to check that Fastapi instrumentation hasn't already unpatched these
+    if isinstance(starlette.routing.Route.handle, ObjectProxy):
+        _u(starlette.routing.Route, "handle")
+
+    if isinstance(starlette.routing.Mount.handle, ObjectProxy):
+        _u(starlette.routing.Mount, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    # Since handle can be called multiple times for one request, we take the path of each instance
+    # Then combine them at the end to get the correct resource name
+    scope = get_argument_value(args, kwargs, 0, "scope")
+
+    if "__dd_paths__" in scope:
+        scope["__dd_paths__"].append(instance.path)
+
+    else:
+        scope["__dd_paths__"] = [instance.path]
+
+    if "datadog" not in scope:
+        return wrapped(*args, **kwargs)
+
+    # The scope["datadog"] object gets replaced with each new call to the handler
+    # Therefore we need to check if the current request span is the first one, and if it is, place it in __dd__
+    # So that it's not replaced and can be given the correct resource name
+    if "__dd__" not in scope:
+        scope["__dd__"] = {"__dd_first_request_span__": scope["datadog"].get("request_span")}
+
+    request_span = scope["__dd__"].get("__dd_first_request_span__")
```

Comment:
Can we put this in an `else` so that we don't have to dig out the span we just set here?